### ISSUE TITLE
Problem when generating from enum type of protobuf

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#270]
_Originally opened by @ryoya-fujimoto in https://github.com/cuelang/cue/issues/270_

Generated cue file from protobuf, enum in protobuf trancerate to numeric values like this.

proto
```
enum Status {
  STATUS_UNKNOWN = 0;
  STATUS_CONNECTED = 1;
  STATUS_DISCONNECTED = 2;
}
```

generated cue
```
Status: "STATUS_UNKNOWN" | "STATUS_CONNECTED" | "STATUS_DISCONNECTED"
Status_value: {
	STATUS_UNKNOWN:        0
	STATUS_CONNECTED:      1
	STATUS_DISCONNECTED:   2
}
```

If `Status` is not used in cue, cue cannot be compiled with a `incomplete value` error. This is inconvenient because not all types generated from protobuf are always used.
So how about setting a default value for cue generated as shown below?

```
Status: *"STATUS_UNKNOWN" | "STATUS_CONNECTED" | "STATUS_DISCONNECTED"
```

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/issues/270#issuecomment-580331126_

Can you say how to reproduce the problem?

In general, enums are used in messages, where they are optional. So such messages should not be incomplete.
If you mean evaluating the produced config, the real issue here is that the messages, enums etc. should all be definitions. Then this wouldn't be an issue.

In general, it seems that a best practice is for top-level definitions to not have default values. On top of that, all fields in proto3 are optional. But maybe it should generate defaults as prescribed by proto3 for all fields. This generally results in less nice schema, though, and in some cases carrying over this semantics is undesirable. It could be made an option perhaps.

**Comment by cueckoo** [comment]
_Original reply by @ryoya-fujimoto in https://github.com/cuelang/cue/issues/270#issuecomment-581001129_

Thank you for your comment. A simple reproduction procedure is described below.

### 1. create app.proto

```
syntax = "proto3";

message Device {
  string name = 0;
  DeviceType type = 1;
}

enum DeviceType {
  DEVICE_TYPE_UNKNOWN = 0;
  DEVICE_TYPE_PRIMARY = 1;
  DEVICE_TYPE_SECONDARY = 2;
}
```

### 2. Generate cue file using import command

```bash
$ cue import -I ./ ./app.proto
$ cat ./app.proto.cue
Device: {
	name?: string     @protobuf(0)
	type?: DeviceType @protobuf(1)
}
DeviceType: "DEVICE_TYPE_UNKNOWN" |
	"DEVICE_TYPE_PRIMARY" |
	"DEVICE_TYPE_SECONDARY"

DeviceType_value: {
	"DEVICE_TYPE_UNKNOWN":   0
	"DEVICE_TYPE_PRIMARY":   1
	"DEVICE_TYPE_SECONDARY": 2
}
```

### 3. Create `user.cue`

(This step is not necessary to reproduce, but it is listed as a usecase.)

```bash
$ cat << EOS > user.cue
device: Device & {
  name: "device-name"
  type: "DEVICE_TYPE_UNKNOWN"
}
EOS
```

### 4. Export or validate

Want to export or validate `user.cue` with `app.proto.cue`, but it is incomplete error.

```bash
$ cue vet -c app.proto.cue user.cue
DeviceType: incomplete value (("DEVICE_TYPE_UNKNOWN" | "DEVICE_TYPE_PRIMARY" | "DEVICE_TYPE_SECONDARY")):
    ./app.proto.cue:5:13

$ cue export app.proto.cue user.cue
cue: marshal error at path DeviceType: cannot convert incomplete value "(\"DEVICE_TYPE_UNKNOWN\" | \"DEVICE_TYPE_PRIMARY\" | \"DEVICE_TYPE_SECONDARY\")" to JSON
```

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/issues/270#issuecomment-762664112_

Sorry for coming back so late, but I think this is fixed.

The problem was that `vet -c` checks _all_ values, including, in this case, the generated types. As of v0.3.0 proto types are now translated as definitions. This should exclude them from a `vet -c`.

**Comment by cueckoo** [comment]
_Original reply by @myitcv in https://github.com/cuelang/cue/issues/270#issuecomment-809494173_

This appears to be resolved for me, testing against [`v0.3.0-beta.7`](https://github.com/cuelang/cue/releases/tag/v0.3.0-beta.7):

```
exec cue import -I ./ ./app.proto
exec cue vet -c app.cue user.cue
exec cue export app.cue user.cue
cmp stdout stdout.golden

-- app.proto --
syntax = "proto3";

message Device {
  string name = 0;
  DeviceType type = 1;
}

enum DeviceType {
  DEVICE_TYPE_UNKNOWN = 0;
  DEVICE_TYPE_PRIMARY = 1;
  DEVICE_TYPE_SECONDARY = 2;
}
-- user.cue --
device: #Device & {
	name: "device-name"
	type: "DEVICE_TYPE_UNKNOWN"
}
-- stdout.golden --
{
    "device": {
        "name": "device-name",
        "type": "DEVICE_TYPE_UNKNOWN"
    }
}
```

So I will close for now, but @ryoya-fujimoto please say if you are still seeing issues. Thanks
